src/utils/tableDataUtils.ts

```diff
@@ -0,0 +1,99 @@
+import { getOffset } from '@/helpers'
+
+interface TableDataParams {
+  getSingleEntity: (mesh: string, query: string | null, params: { size: number; offset: string | null }) => any
+  getAllEntities: (params: { size: number; offset: string | null }) => any
+  getAllEntitiesFromMesh: (mesh: string) => any
+  mesh: string
+  size: number
+  query: string | null
+  offset: string | null
+}
+
+function sortEntities(
+  items: {
+    name: string
+    mesh: string
+  }[],
+) {
+  return [...items].sort((a, b) => (a.name > b.name ? 1 : a.name === b.name ? (a.mesh > b.mesh ? 1 : -1) : -1))
+}
+
+const getItems = (response: {
+  total: number
+  items: {
+    name: string
+    mesh: string
+  }[]
+}) => {
+  const r = response
+
+  if (!r.total) {
+    return [r]
+  }
+
+  if (r.total !== 0 && r.items && r.items.length > 0) {
+    return sortEntities(r.items)
+  }
+
+  return null
+}
+
+async function getAPICallFunction({
+  getSingleEntity,
+  getAllEntities,
+  getAllEntitiesFromMesh,
+  mesh,
+  query,
+  size,
+  offset,
+}: TableDataParams) {
+  const params = {
+    size,
+    offset,
+  }
+
+  if (mesh === 'all') {
```

Comment:
Do we need an await here? We can just return promise.

Reply:
Good point, we don't need await there